### PR TITLE
Add features: Silent Pod and Pod Diagnostics

### PR DESCRIPTION
**Pull request by marionbarker**
This PR adds the Silent Pod and Pod Diagnostics features to the OmniKit screen for Eros pods:

The Silent Pod and Pod Diagnostics rows are added as shown in the graphic below.

User-facing changes:
* Pod notifications include an option to silence the pod
   * When silenced, alerts are only on the phone, the pod does not beep
   * Pod faults, end of life and out of insulin still alarm on the pod regardless of this setting
* The previous pod row is moved under the pod details row to make it easier to find
* A pod diagnostics row is added to the end of the pod screen
   * various diagnostics rows are added in a new sub-screen
   * Eros pod users can check their pod RSSI to assist in debugging pump communication issues

The before (left) and after (right) screenshots of the Omnipod screen is shown below.

omnikit-pr-32

### CODE REVIEW

File: OmniKitUI/Views/OmnipodSettingsView.swift [hunk 1]
```diff
@@ -441,15 +457,27 @@ struct OmnipodSettingsView: View  {
...
                     HStack {
-                        FrameworkLocalText("Confidence Reminders", comment: "Text for confidence reminders navigation link").foregroundColor(Color.primary)
+                        FrameworkLocalText("Confidence Reminders", comment: "Text for confidence reminders navigation link")
+                            .foregroundColor(Color.primary)
                         Spacer()
                         Text(viewModel.beepPreference.title)
                             .foregroundColor(.secondary)
                     }
                 }
+                if  (allowDebugFeatures) {
```
Comment (itsmojo) [inline OmniKitUI/Views/OmnipodSettingsView.swift]: Extra space after the "if" keyword.

I'd much prefer that Silence Pod support not be hidden if allowDebugFeatures isn't true as it's a very useful basic feature that should always be available. Also, if it is sometimes hidden and other times not, then if the OmniBLEPumpManagerState.silencePod state variable gets set to true while allowDebugFeatures is true, then this state variable can never be reset by the user and all future pods will never give audible pod alerts if subsequent versions of Loop don't also have allowDebugFeatures set to true.

File: OmniKitUI/Views/OmnipodSettingsView.swift [hunk 2]
```diff
@@ -489,14 +517,17 @@ struct OmnipodSettingsView: View  {
                 }
             }
 
-            if let previousPodDetails = viewModel.previousPodDetails {
+            if  (allowDebugFeatures) {
```
Comment (itsmojo) [inline OmniKitUI/Views/OmnipodSettingsView.swift]: Extra space after the "if" keyword.

**Review by itsmojo — COMMENTED** [review]
LGTM except for extra spaces in allowDebugFeatures if statements and a big concern about using allowDebugFeatures to conditionally enable displaying the Silence Pod option which I believe should always be available. If this option really must be conditional like this, additional thought and code will be needed to properly handle certain possible cases.

File: OmniKitUI/Views/OmnipodSettingsView.swift (hunk 1 above)
Comment (marionbarker) [reply OmniKitUI/Views/OmnipodSettingsView.swift]: I was thinking of the requirement needed with LoopKit / MockKit which is also used by Tidepool. But the DIY pump managers will not be shared with Tidepool. So I reverted the commit that added the flag.

**Comment by marionbarker** [comment]
The DIY pump managers will not be used by Tidepool, so I reverted the commit that added the allowDebugFeatures check.

**Comment by marionbarker** [comment]
Graphics for the modified rows and new features are reporting in the OmniBLE PR.

For modified rows:
* https://github.com/LoopKit/OmniBLE/pull/118#issuecomment-2035127532

For new feature: Silent Pod:
* https://github.com/LoopKit/OmniBLE/pull/118#issuecomment-2035185879

For new feature: Pod Diagnostics:
* https://github.com/LoopKit/OmniBLE/pull/118#issuecomment-2035201631

**Review by ps2 — APPROVED** [review]
LGTM!

**Review by ps2 — CHANGES_REQUESTED** [review]
asyncAction and FirstAppear are wrongly pushing view model logic into the view.

**Comment by marionbarker** [comment]
This PR has been updated as requested and is ready for final review.

**Review by itsmojo — APPROVED** [review]

**Pull request merged**